Tasks/CondaEnvironmentV0/conda.ts

```diff
@@ -7,13 +7,28 @@ import * as internal from './conda_internal';
 import { Platform } from './taskutil';
 
 interface TaskParameters {
-    environmentName: string,
+    customEnvironment?: boolean,
+    environmentName?: string,
     packageSpecs?: string,
     updateConda?: boolean,
+    installOptions?: string,
     createOptions?: string,
     cleanEnvironment?: boolean
 }
 
+// TODO copied from PythonScript. Find a way to share between tasks.
+/**
+ * Check for a parameter at runtime.
+ * Useful for conditionally-visible, required parameters.
+ */
+function assertParameter<T>(value: T | undefined, propertyName: string): T {
```

Comment:
tl.getInput has a required or not required check. Do you need this beyond that?

Reply:
Yes: if a field is conditionally visible, it can be "required" by the function but still `undefined` in TypeScript.  So this is saying "I know that the condition for the parameter to be required is fulfilled."

Reply:
Or rather, conditionally visible inputs can be "required" in the task.json (they get the red asterisk in the UI and you can't queue a build without them), but if you "require" them in `getInput`, the task will fail at runtime if they are being hidden.  So this is a delayed "require" check.

Reply:
In the past we have generally given a default value in task.json. This is interesting and if it is a common scenario should be an override on the getInput method in the task-lib.

Reply:
`environmentName` has a default value in the task.json of `""`, but it probably needs a truthy value.

I'm looking at other tasks and they just don't bother to check for `undefined` if the condition is fulfilled.  The reason I need this here is because I'm building the task with `--strictNullChecks`, so everything that can possibly be null or undefined has to be explicitly handled.